code/datums/gamemode/factions/bloodcult/bloodcult_mobs_and_constructs.dm

```diff
@@ -0,0 +1,439 @@
+
+/////////////////Juggernaut///////////////
+/mob/living/simple_animal/construct/armoured/perfect
+	icon_state = "juggernaut2"
+	icon_living = "juggernaut2"
+	icon_dead = "juggernaut2"
+	construct_spells = list(
+		/spell/aoe_turf/conjure/forcewall/greater,
+		/spell/juggerdash,
+		)
+	see_in_dark = 7
+	var/dash_dir = null
+	var/turf/crashing = null
+
+/mob/living/simple_animal/construct/armoured/perfect/New()
+	..()
+	setupfloat()
+
+/mob/living/simple_animal/construct/armoured/perfect/to_bump(var/atom/obstacle)
+	if(src.throwing)
+		var/breakthrough = 0
+		if(istype(obstacle, /obj/structure/window/))
+			obstacle.Destroy(brokenup = 1)
+			breakthrough = 1
+
+		else if(istype(obstacle, /obj/structure/grille/))
+			var/obj/structure/grille/G = obstacle
+			G.health = (0.25*initial(G.health))
+			G.broken = 1
+			G.icon_state = "[initial(G.icon_state)]-b"
+			G.setDensity(FALSE)
+			getFromPool(/obj/item/stack/rods, get_turf(G.loc))
+			breakthrough = 1
+
+		else if(istype(obstacle, /obj/structure/table))
+			var/obj/structure/table/T = obstacle
+			T.destroy()
+			breakthrough = 1
+
+		else if(istype(obstacle, /obj/structure/rack))
+			new /obj/item/weapon/rack_parts(obstacle.loc)
+			qdel(obstacle)
+			breakthrough = 1
+
+		else if(istype(obstacle, /turf/simulated/wall))
+			var/turf/simulated/wall/W = obstacle
+			if (W.hardness <= 60)
+				playsound(W, 'sound/weapons/heavysmash.ogg', 75, 1)
+				W.dismantle_wall(1)
+				breakthrough = 1
+			else
+				src.throwing = 0
+				src.crashing = null
+
+		else if(istype(obstacle, /obj/structure/reagent_dispensers/fueltank))
+			obstacle.ex_act(1)
+
+		else if(istype(obstacle, /mob/living))
+			var/mob/living/L = obstacle
+			if (L.flags & INVULNERABLE)
+				src.throwing = 0
+				src.crashing = null
+			else if (!(L.status_flags & CANKNOCKDOWN) || (M_HULK in L.mutations) || istype(L,/mob/living/silicon))
+				//can't be knocked down? you'll still take the damage.
+				src.throwing = 0
+				src.crashing = null
+				L.take_overall_damage(5,0)
+				if(L.locked_to)
+					L.locked_to.unlock_atom(L)
+			else
+				L.take_overall_damage(5,0)
+				if(L.locked_to)
+					L.locked_to.unlock_atom(L)
+				L.Stun(5)
+				L.Knockdown(5)
+				L.apply_effect(STUTTER, 5)
+				playsound(src, 'sound/weapons/heavysmash.ogg', 50, 0, 0)
+				breakthrough = 1
+		else
+			src.throwing = 0
+			src.crashing = null
+
+		if(breakthrough)
+			if(crashing && !istype(crashing,/turf/space))
+				spawn(1)
```

Comment:
why the spawning? Dramatic effect, or something going wrong that needs it to be desynced?

Reply:
throw_at contains a loop that doesn't end until the throwing is over. just like when firing a projectile, you should always place a spawn beforehand. The (1) gives some hit stop which helps feeling the obstacle you're punching through.